projects/igniteui-angular/src/lib/input-group/input-group.component.ts

```diff
@@ -200,48 +187,18 @@ export class IgxInputGroupComponent extends DisplayDensityBase implements IgxInp
 
     /**
      * An @Input property that sets how the input will be styled.
-     * The allowed values are `line`, `box`, `border`, `fluent`, `search`, `fluentSearch`, `bootstrap` and `indigo`. The default is `line`.
+     * Allowed values of type IgxInputGroupType.
      * ```html
      * <igx-input-group [type]="'search'">
      * ```
      */
     @Input('type')
-    public set type(value: string) {
-        const type: IgxInputGroupType = (IgxInputGroupType as any)[value.toUpperCase()];
-        if (type !== undefined) {
-            this.isBox = this.isBootstrap = this.isFluent = this.isIndigo = this.isFluentSearch = this.isBorder = this.isSearch = false;
-            switch (type) {
-                case IgxInputGroupType.BOX:
-                    this.isBox = true;
-                    break;
-                case IgxInputGroupType.BORDER:
-                    this.isBorder = true;
-                    break;
-                case IgxInputGroupType.FLUENT:
-                    this.isFluent = true;
-                    break;
-                case IgxInputGroupType.INDIGO:
-                    this.isIndigo = true;
-                    break;
-                case IgxInputGroupType.FLUENT_SEARCH:
-                    this.isFluentSearch = true;
-                    break;
-                case IgxInputGroupType.SEARCH:
-                    this.isSearch = true;
-                    break;
-                case IgxInputGroupType.BOOTSTRAP:
-                    this.isBootstrap = true;
-                    break;
-                default: break;
-            }
-
-            this._type = type;
-        }
+    public set type(value: IgxInputGroupType) {
```

Comment:
Minor thing (can be ignored): if this doesn't require logic anymore, it could be just a normal property with the default value assigned :)

PS: Looking at the description & code - do I see right or did `fluentSearch` never quite worked before?

Reply:
Might be. I am looking forward to further reducing the amount of types allowed to just - box, border, line, and search. There's a lot of work to be done on the styling side for this to happen.